packages/app/src/files-manager/components/files-manager/files-manager.component.tsx

```diff
@@ -197,16 +198,21 @@ const FilesManager: FunctionComponent<FilesManagerProps> = ({
     }
   }, [resetUploadingState])
 
-  const onFileInputChange = () => {
-    if (fileInputRef.current?.files?.length) {
-      void dispatchThunk(
-        uploadFile(
-          Array.from(fileInputRef.current?.files).map((file) => file.path)
-        )
-      )
-      fileInputRef.current.value = ""
+  useEffect(() => {
+    if (uploadActionTrigger) {
+      const uploadActionPromise = dispatchThunk(uploadFile())
+
+      //abort on dismount or when a new upload has been triggered
+      return () => {
+        if ("abort" in uploadActionPromise) {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-explicit-any
+          ;(uploadActionPromise as any).abort()
+        }
+      }
     }
-  }
+    return noop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [uploadActionTrigger])
```

Comment:
Can we catch up tomorrow and talk about this? Seems to me like a working-version with this any type and searching key in promise.